### PR TITLE
Use remix plugin 0.3

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: apps/remix-ide/package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "yo-yoify": "^3.7.3"
   },
   "dependencies": {
-    "@remixproject/engine": "^0.2.5",
+    "@remixproject/engine-web": "^0.2.5",
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/package.json:83]: Should be 0.3.0

**Review by GrandSchtroumpf — COMMENTED** [review]
In PluginManager rename : 
- `canActivate` to `canActivatePlugin`
- `canDeactivate` to `canDeactivatePlugin`

File: apps/remix-ide/src/app/components/plugin-manager-component.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ const yo = require('yo-yo')
 const csjs = require('csjs-inject')
 const EventEmitter = require('events')
 const LocalPlugin = require('./local-plugin')
-import { ViewPlugin, IframePlugin, WebsocketPlugin } from '@remixproject/engine'
+import { IframePlugin, WebsocketPlugin } from '@remixproject/engine-web'
+import { ViewPlugin } from '@remixproject/engine'
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/app/components/plugin-manager-component.js:6]: In 0.3.3 this should come from engine-web too

File: apps/remix-ide/src/app/tabs/debugger-tab.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ class DebuggerTab extends ViewPlugin {
 
     this.renderComponent()
 
-    this.call('manager', 'activatePlugin', 'source-verification').catch(e => console.log(e.message))
+    // this.call('manager', 'activatePlugin', 'source-verification').catch(e => console.error(e))
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/app/tabs/debugger-tab.js:64]: You can remove

File: apps/remix-ide/src/remixAppManager.js [hunk 4]
```diff
@@ -28,11 +29,13 @@ export class RemixAppManager extends PluginManager {
     this.permissionHandler = new PermissionHandler()
   }
 
-  async canActivate (from, to) {
-    return canActivate(from.name)
+  async canActivatePlugin (from, to) {
+    return true // canActivate(from.name)
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/remixAppManager.js:33]: any plugin can activate another ?

File: apps/remix-ide/src/remixAppManager.js [hunk 5]
```diff
@@ -28,11 +29,13 @@ export class RemixAppManager extends PluginManager {
...
   }
 
-  async canDeactivate (from, to) {
+  async canDeactivatePlugin (from, to) {
+    console.log('canDeactivatePlugin', from, to)
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/remixAppManager.js:37]: Maybe remove console.log

File: apps/remix-ide/src/remixAppManager.js [hunk 6]
```diff
@@ -69,8 +72,8 @@ export class RemixAppManager extends PluginManager {
     this.event.emit('deactivate', plugin)
   }
 
-  onRegistration (plugin) {
-    this.event.emit('added', plugin.name)
+  onRegistration () {
+    console.log('manager has been registered')
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/remixAppManager.js:76]: same here

File: apps/remix-ide/src/remixAppManager.js [hunk 7]
```diff
@@ -105,8 +103,12 @@ export class RemixAppManager extends PluginManager {
...
       'location': 'mainPanel'
-    })
-    return plugins.map(plugin => new IframePlugin(plugin))
+    })    
+    return plugins.map(plugin => {
+      if (plugin.name === 'scriptRunner') {
+        plugin.url = 'ipfs://QmRWBCgCzKa1ixoVN9mvJACVuqWLRTRYS4sUJ9wzUBRHSf'
```
Comment (GrandSchtroumpf) [inline apps/remix-ide/src/remixAppManager.js:109]: Why not setting it directly in the profile above ?

File: libs/remixd/package.json [hunk 8]
```diff
@@ -32,10 +32,10 @@
   },
   "homepage": "https://github.com/ethereum/remix-project#readme",
   "dependencies": {
-    "@remixproject/plugin": "0.3.0-beta.5",
-    "@remixproject/plugin-api": "0.3.0-beta.5",
-    "@remixproject/plugin-utils": "0.3.0-beta.5",
-    "@remixproject/plugin-ws": "^0.3.0-beta.8",
+    "@remixproject/plugin": "^0.3.2",
```
Comment (GrandSchtroumpf) [inline libs/remixd/package.json:35]: use version 0.3.3

**Review by GrandSchtroumpf — APPROVED** [review]

File: workspace.json [hunk 9]
```diff
@@ -104,7 +104,7 @@
           "builder": "@nrwl/workspace:run-commands",
           "options": {
             "commands": [
-              "node_modules/.bin/tsc -p apps/remix-ide-e2e/tsconfig.e2e.json"
+              "tsc -p apps/remix-ide-e2e/tsconfig.e2e.json"
```
Comment (cleytontiagosorinho) [inline workspace.json:107]: Sim

File: libs/remixd/package.json [hunk 10]
```diff
@@ -35,10 +35,10 @@
   },
   "homepage": "https://github.com/ethereum/remix-project#readme",
   "dependencies": {
-    "@remixproject/plugin": "0.3.0-beta.5",
-    "@remixproject/plugin-api": "0.3.0-beta.5",
-    "@remixproject/plugin-utils": "0.3.0-beta.5",
-    "@remixproject/plugin-ws": "^0.3.0-beta.8",
+    "@remixproject/plugin": "^0.3.3",
```
Comment (cleytontiagosorinho) [inline libs/remixd/package.json:38]: 7139bd9c1eb73cd1b6ebff99f500b431a987f706

**Review by cleytontiagosorinho — CHANGES_REQUESTED** [review]

**Review by cleytontiagosorinho — APPROVED** [review]
7139bd9c1eb73cd1b6ebff99f500b431a987f706

**Pull request merged**